src/test/crimson/test_config.cc

```diff
@@ -1,26 +1,75 @@
 #include <chrono>
+#include <string>
 #include <numeric>
 #include <seastar/core/app-template.hh>
 #include <seastar/core/sharded.hh>
+#include "common/config_obs.h"
 #include "crimson/common/config_proxy.h"
 
 using Config = ceph::common::ConfigProxy;
+const std::string test_uint_option = "osd_max_pgls";
+
+class ConfigObs : public ceph::internal::md_config_obs_impl<Config> {
+  uint64_t last_change;
+  uint64_t num_changes = 0;
+  std::unique_ptr<const char*[]> conf_keys = std::make_unique<const char*[]>(2);
+
+  const char** get_tracked_conf_keys() const override {
+    return conf_keys.get();
+  }
+  void handle_conf_change(const Config& conf,
+                          const std::set <std::string> &changes) override{
+    if (changes.count(test_uint_option)) {
+      last_change = conf.get_val<uint64_t>(test_uint_option);
+      num_changes += 1;
+    }
+  }
+public:
+  ConfigObs() {
+    conf_keys[0] = test_uint_option.c_str();
+    conf_keys[1] = NULL;
+    ceph::common::local_conf().add_observer(this);
+  }
+
+  uint64_t get_last_change() const { return last_change; }
+  uint64_t get_num_changes() const { return num_changes; }
+  seastar::future<> stop() {
+    ceph::common::local_conf().remove_observer(this);
+    return seastar::now();
+  }
+};
+
+seastar::sharded<ConfigObs> sharded_cobs;
 
 static seastar::future<> test_config()
 {
   return ceph::common::sharded_conf().start().then([] {
     return ceph::common::sharded_conf().invoke_on(0, &Config::start);
   }).then([] {
-    return ceph::common::sharded_conf().invoke_on_all([](Config& config) {
-      return config.set_val("osd_tracing", "true");
-    });
+    return sharded_cobs.start();
   }).then([] {
     return ceph::common::sharded_conf().invoke_on_all([](Config& config) {
-      if (!config.get_val<bool>("osd_tracing")) {
-        throw std::runtime_error("run osd_tracing");
-      }
+      return config.set_val(test_uint_option,
+                            std::to_string(seastar::engine().cpu_id()));
     });
   }).then([] {
+    return seastar::do_with(ceph::common::local_conf().get_val<uint64_t>(test_uint_option),
+                            [](auto expected) {
```

Comment:
the `do_with()` part isn't necessary here - internally it wraps the first argument in a `unique_ptr` so its memory stays around until completion. but since this value is just a `uint64_t`, we can copy it into the lambda captures without worrying about lifetime issues. replace this call with `auto expected = ceph::common::local_conf()...` and the rest should work fine